src/utils.cpp

```diff
@@ -766,5 +766,59 @@ bool executePreAction(const nlohmann::json& i_parsedConfigJson,
 
     return true;
 }
+
+std::string getFRUPathFromJSON(const nlohmann::json i_sysCfgJsonObj,
+                               const std::string& i_vpdPath)
+{
+    if (i_vpdPath.empty())
+    {
+        throw std::runtime_error("Empty file path provided.");
+    }
+
+    if (!i_sysCfgJsonObj.contains("frus"))
+    {
+        throw std::runtime_error("Missing frus section in VPD JSON");
+    }
+
+    // If given path is FRU EEPROM path
+    if (i_sysCfgJsonObj["frus"].contains(i_vpdPath))
+    {
+        return i_vpdPath;
+    }
+
+    const nlohmann::json& l_fruList =
+        i_sysCfgJsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& l_fru : l_fruList.items())
+    {
+        const auto l_fruPath = l_fru.key();
+
+        // If given path is redundant EEPROM path
+        if (i_vpdPath == i_sysCfgJsonObj["frus"][l_fruPath].at(0).value(
```

Comment:
have an issue on "at(0)"

Reply:
we decided to hardcode at(0), as the json structure is fixed to have common tags ( like redundantEeprom ) in first inventory item only which acts as the parent fru.